src/service/exception_handlers.py

```diff
@@ -0,0 +1,70 @@
+"""
+Exception handlers for the FastAPI application.
+"""
+
+import logging
+
+from fastapi import HTTPException, Request, status
+from fastapi.responses import JSONResponse
+
+from src.service.error_mapping import map_error
+from src.service.exceptions import SparkManagerError
+
+logger = logging.getLogger(__name__)
+
+
+def _format_error(
+    status_code: int, error_code: int, error_type_str: str, message: str | None
+):
+    """Format error response with consistent structure."""
+    return JSONResponse(
+        status_code=status_code,
+        content={
+            "error": error_code,
+            "error_type": error_type_str,
+            "message": message or error_type_str,
+        },
+    )
+
+
+async def universal_error_handler(request: Request, exc: Exception):
+    """
+    Universal handler for all types of exceptions.
+
+    Handles:
+    - SparkManagerError and its subclasses (including all specific error types)
+    - HTTPException from FastAPI
+    - Generic exceptions
+    """
+    # Default values
+    error_code = 0
+    error_type_str = "Unknown error"
+    message = None
```

Comment:
these 2 are set in the 3 blocks below and so don't need to be initialized here

Reply:
👍

Reply:
Looks like you missed message?

Reply:
No I removed the assignment.

Reply:
Oh sorry I means error_type_str

Reply:
👍